sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/StorageOutputStream.java

```diff
@@ -0,0 +1,155 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.common;
+
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.util.annotation.NonNull;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * StorageOutputStream
+ */
+public abstract class StorageOutputStream extends OutputStream {
+    final ClientLogger logger = new ClientLogger(StorageOutputStream.class);
+
+    /*
+     * Holds the write threshold of number of bytes to buffer prior to dispatching a write. For block blob this is the
+     * block size, for page blob this is the Page commit size.
+     */
+    final int writeThreshold;
```

Comment:
protected

Reply:
This is only used in StorageOutputStream, all child class will pass in the writeThreshold to base class

Reply:
I made it private.